src/pool/srv_pool.c

```diff
@@ -4420,9 +4476,71 @@ static int ds_pool_mark_upgrade_completed(uuid_t pool_uuid,
 	ABT_rwlock_unlock(svc->ps_lock);
 	rdb_tx_end(&tx);
 out:
+	D_DEBUG(DB_MD, DF_UUID "mark upgrade complete.: %d/%d\n", DP_UUID(pool_uuid), rc1, rc);
 	return rc1;
 }
 
+/* check and upgrade the object layout if needed.
+ * rc = 0 does not need upgrade.
+ * rc = 1 schedule upgrade job.
+ * other  failure.
+ */
+static int
+pool_check_upgrade_object_layout(struct rdb_tx *tx, struct pool_svc *svc)
+{
+	d_iov_t		value;
+	uint32_t	current_layout_ver = 0;
+	int		rc = 0;
+
+	d_iov_set(&value, &current_layout_ver, sizeof(current_layout_ver));
+	rc = rdb_tx_lookup(tx, &svc->ps_root, &ds_pool_prop_obj_version, &value);
+	if (rc && rc != -DER_NONEXIST)
+		return rc;
+	else if (rc == -DER_NONEXIST)
+        	current_layout_ver = 0;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs